miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -178,6 +180,69 @@ export.set_account_item
     # => [R', V]
 end
 
+#! Returns VALUE located under specified KEY in map in specified account storage slot. 
+#! Panics if 
+#! - the index is out of bounds.
+#! - the requested storage slot type is not map
+#!
+#! Stack: [index, KEY, ...]
+#! Output: [VALUE]
+#!
+#! - index is the index of the item to get.
+#! - VALUE is the value of the item.
+export.get_account_map_item
+    # check if storage type is map
+    dup exec.account::get_storage_slot_type_info drop
+    # => [slot_type, index, KEY, ...]
+
+    # fails if slot_type is not 1 = map
+    assert
+    # => [index, KEY, ...]
+    
+    # fetch the account storage item, which is ROOT of the map
+    exec.account::get_item swapw
+    # => [KEY, ROOT ...]
+
+    # fetch the VALUE located under KEY in the tree
+    exec.smt::get 
+    # => [VALUE, ROOT, ...]
+
+    # organize the stack for return
+    swapw dropw
+    # => [VALUE]
+end
+
+#! Inserts specified NEW_VALUE under specified KEY in map in specified account storage slot. 
+#! Panics if 
+#! - the index is out of bounds.
+#! - the requested storage slot type is not map
+#!
+#! Stack: [index, NEW_KEY, NEW_VALUE, ...]
+#! Output: [OLD_VALUE]
+#!
+#! - index is the index of the item to get.
+#! - NEW_VALUE is the value of the new item.
+export.set_account_map_item
+    # check if storage type is map
+    dup exec.account::get_storage_slot_type_info drop
+    # => [slot_type, index, NEW_KEY, NEW_VALUE, ...]
+
+    # fails if slot_type is not 1 = map
+    assert
+    # => [index, NEW_KEY, NEW_VALUE, ...]
```

Comment:
question: we will never allow the storage type to change?

Reply:
We can change the storage type as well. But not using this procedure. We would need to use `export.set_account_item` and use the index of the storage map

Reply:
I think for now we can assume that storage layout (i.e., type of each storage slot) can be defined only at account creation time (this would simplify quite a few things). We can relax this condition some time down the road.